docs/content/migration.md

```diff
@@ -0,0 +1,545 @@
+use normal Angular class bindings to toggle classes like
+`[class.my-dark-theme]="isDarkMode"`.
+
+#### Applying theming to custom components
+
+AngularJS Material's [`md-colors`][73] directive and [`$mdColors`][125] service support looking up
+and [dynamically applying `theme-palette-hue-opacity` color combinations][73] as CSS properties to
+your application's custom components.
+
+Angular Material uses the [`mat-color`][74] Sass function to lookup theme colors and 
+Sass mixins with CSS selectors (see `super-special-button` example of a class above) to
+[style application or library components][168].
+
+##### Accessing Hues
+
+AngularJS Material's [Hue classes][169] (`md-hue-1`, `md-hue-2`, `md-hue-3`) are used to modify a
+component's hues.
+
+These classes are not supported out of the box with Angular Material, but you can use the
+[`mat-color`][74] Sass function to lookup the desired Hues and assign them to all of the appropriate
+CSS selectors and CSS properties, for a specific component, in a theme mixin. 
+
+#### Browser header and system status bar coloring on mobile
+
+AngularJS Material's theme system supports modifying the
+[browser header and system status bar colors][149].
+
+Angular's [Meta service][147] can be used to dynamically modify `<meta>` tags that affect browser
+header and status bar colors. The [AngularJS Material docs][149] for this feature have links to
+learn more about the specific `<meta>` tags.
+
+##### Note about CSS variables
+Once the team begins phasing out support for IE11, the theming system will be revisited with
+[CSS Variables][160] (also known as CSS Custom Properties) in mind.
+
+### <a name="layout"></a> Changes to Layout Features
+
+The AngularJS Material [layout][108] system includes includes attributes and classes like `layout`,
+`flex`, `show`, `hide`, etc.
+
+#### <a name="flex-layout"></a> Flex Layout
+
+This comprehensive layout system was moved into its [own project][120] (beta).
+
+The package contains the following:
+- A [Declarative API][116] that includes attributes like `fxLayout`, `fxFlex`, `fxHide`, `fxShow`,
+  and more.
+- A [Responsive API][117] that includes attributes like `fxLayout.lt-md`, `fxFlex.gt-sm`,
+  `fxHide.gt-xs`, `[ngClass.sm]`, `[ngStyle.xs]`, and more.
+- Both APIs include [CSS Grid][65] features in addition to the [CSS Flexbox][flexbox] features from
+  AngularJS Material.
+- The ability to define [Custom Breakpoints][118]
+- A [MediaObserver service][119] that allows subscribing to `MediaQuery` activation changes
+
+Since AngularJS Material's inception, browser support for [CSS Flexbox][flexbox] has [broadly
+improved][177] and [CSS Grid][65] has become [widely available][178]. Application developers should
+carefully consider the the costs of additional run-time dependencies versus native alternatives
+when choosing an approach to layout. This library adds a nontrivial payload cost.
```

Comment:
Andrew brought this up:

What do you think about adding a bit more here like
```markdown
Many of the AngularJS Material layout APIs map to straightforward, vanilla CSS using Flexbox
and media queries. We recommend checking out [the MDN docs](https://developer.mozilla.org/en-US/docs/Learn/CSS/CSS_layout/Flexbox),
[this scotch.io guide](https://scotch.io/tutorials/a-visual-guide-to-css3-flexbox-properties),
and [Flexbox Froggy](https://flexboxfroggy.com).

```

Reply:
Yeah, I think that we can certainly flesh this out. I think that providing some links to learning resources will be good as well since many AngularJS Material developers have spent years not needing to learn the specifics of CSS Flexbox and CSS Grid.